index.js

```diff
@@ -23,15 +23,17 @@ function nGram(n) {
       return nGrams;
     }
 
-    value = String(value);
+    if (!Array.isArray(value)) {
+      value = String(value);
+    }
     index = value.length - n + 1;
 
     if (index < 1) {
       return nGrams;
     }
 
     while (index--) {
-      nGrams[index] = value.substr(index, n);
+      nGrams[index] = Array.isArray(value) ? value.slice(index, index + n) : value.substr(index, n);
```

Comment:
I was thinking of using `.slice` both on [Array](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice) and on [String](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/slice) here, would that work?